python/lsst/dax/imgserv/MetadataFitsDb.py

```diff
@@ -320,47 +308,40 @@ def insertMetadataFits(self, metadata):
                         for v in value:
                             # scanFile should be skipping line numbers when it sees a tuple,
                             # so that there are no duplicates.
-                            self._insertFitsValue(cursor, lastFitsFileId, key, v, num, comment)
+                            self._insertFitsValue(lastFitsFileId, key, v, num, comment)
                             num += 1
                     else:
-                        self._insertFitsValue(cursor, lastFitsFileId, key, value, lineNum, comment)
+                        self._insertFitsValue(lastFitsFileId, key, value, lineNum, comment)
                 for hdu in range(1,hdus+1):
-                    metadataPosition = MetadataPosition(lastFitsFileId, hdu, cursor, entries)
+                    metadataPosition = MetadataPosition(lastFitsFileId, hdu, self._conn, entries)
                     metadataPosition._insert()
-                cursor.execute("COMMIT")
-        except MySQLdb.Error as err:
-            cursor.execute("ROLLBACK")
+                trans.commit()
```

Comment:
commit should happen outside `if` (if `len(r)` is >= 1 then you neither commit nor rollback, and I'm not sure what happens in this case).
I strongly recommend to use context management with transactions, this guarantees that transaction is properly closed (or aborted if exception happens). Check third example from http://docs.sqlalchemy.org/en/rel_1_0/core/connections.html#using-transactions